### ISSUE TITLE
No highlighting for filetype nix

### ISSUE THREAD

**Issue by michaelfranzl** [nvim-treesitter/nvim-treesitter#2923]
### Describe the bug

There is no syntax highlighting at all for file type `nix`.

### To Reproduce

Open an existing `.nix` file or create one with the following example contents:

```nix
{ pkgs ? import <nixpkgs> {} }:

pkgs.python3Packages.buildPythonApplication {
  pname = "myapp";
}
```

### Expected behavior

I expect syntax highlighting to work.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - WARNING: tree-sitter executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
  - OK: node found v18.0.0 (only needed for :TSInstallFromGrammar)
  - OK: git executable found.
  - OK: cc executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (Debian 10.2.1-6) 10.2.1 20210110
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - make           ✓ . . . ✓
  - c              ✓ ✓ ✓ ✓ ✓
  - llvm           ✓ . . . .
  - vim            ✓ ✓ . . ✓
  - json           ✓ ✓ ✓ ✓ .
  - rust           ✓ ✓ ✓ ✓ ✓
  - go             ✓ ✓ ✓ ✓ ✓
  - lua            ✓ ✓ ✓ ✓ ✓
  - python         ✓ ✓ ✓ ✓ ✓
  - ruby           ✓ ✓ ✓ ✓ ✓
  - glsl           ✓ ✓ ✓ ✓ ✓
  - dockerfile     ✓ . . . ✓
  - bash           ✓ ✓ ✓ . ✓
  - cpp            ✓ ✓ ✓ ✓ ✓
  - css            ✓ . ✓ ✓ ✓
  - html           ✓ ✓ ✓ ✓ ✓
  - javascript     ✓ ✓ ✓ ✓ ✓
  - hcl            ✓ . ✓ ✓ ✓
  - ninja          ✓ . ✓ ✓ .
  - nix            ✓ ✓ ✓ . ✓
  - scss           ✓ . . ✓ .
  - tsx            ✓ ✓ ✓ ✓ ✓
  - typescript     ✓ ✓ ✓ ✓ ✓
  - yaml           ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.8.0-dev+168-g5359be789
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by user@host

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/home/user/local/share/nvim"

Run :checkhealth for more info
```

### Additional context

To isolate this issue as much as possible, I set `XDG_CONFIG_HOME` to an otherwise empty directory with just one file `init.lua` containing only the following:

```lua
require'nvim-treesitter.configs'.setup {
  -- A list of parser names, or "all"
  ensure_installed = { "c", "lua", "rust", "python", "ruby", "go", "glsl", "dockerfile", "bash", "cpp", "css", "html", "javascript", "json", "hcl", "make", "ninja", "nix", "scss", "tsx", "typescript", "vim", "yaml", "llvm"},
  indent = { enable = false }, -- maybe https://github.com/cstrahan/tree-sitter-nix/issues/17 ?
}
```

`echo &filetype` prints `nix`.

All  of the following quick fix attempts did not change the issue:

```
TSBufEnable highlight nix
TSInstall nix
TSUpdate
```

Maybe this issue is related to https://github.com/cstrahan/tree-sitter-nix/issues/17 ?

**Comment by kyazdani42** [comment]
cannot reproduce, nix highlighting works just fine for me.
Can you add `highlight = { enable = true }` in your setup ?

**Comment by theHamsta** [comment]
Is your nix file recognized by Neovim? Does `:set ft=nix` change anything?

**Comment by michaelfranzl** [comment]
@kyazdani42 Adding `highlight = { enable = true }` made all the difference. It is even in the README: `All modules are disabled by default and need to be activated explicitly`. As a new user I had assumed that  "modules" are "parsers". I did not expect that highlighting would be disabled by default. But what really fooled me was the fact that bare neovim (without plugins, with empty user configuration, i.e. only defaults) already does some kind of base syntax highlighting for example for C and JavaScript, but just not for Nix, which is why I created this issue.

And `TSBufEnable highlight nix` silently didn't work because it's supposed to be `TSBufEnable highlight`, which indeed works.

So, sorry for the confusion and thanks for the help.
